lib/dataset/dataset.cpp

```diff
@@ -277,10 +277,16 @@ typename Masks::holder_type union_or(const Masks &currentMasks,
   for (const auto &[key, item] : otherMasks) {
     if (!currentMasks.contains(key))
       out.insert_or_assign(key, copy(item));
-    else if (out[key].dims().includes(item.dims()))
+    else if (item.dtype() != core::dtype<bool>) {
+      std::string optional_message = " This operation is not supported for "
+                                     "non-boolean masks with same names.";
```

Comment:
Might be more useful to instead include the name of the offending mask, something like:

```
"Cannot combine non-boolean mask '" + key + "' in operation."
```

Reply:
Yes...! I will do.

Reply:
I created a new PR in https://github.com/scipp/scipp/pull/2975